Kinvey/KinveyTests/MultiInsertSpec.swift

```diff
@@ -0,0 +1,1492 @@
+                        expect(result?.errors.count).to(equal(2))
+                        
+                        expect((result?.errors.first as? MultiSaveError)?.index).to(equal(0))
+                        expect((result?.errors.first as? MultiSaveError)?.code).to(equal(11000))
+                        
+                        expect((result?.errors.last as? MultiSaveError)?.index).to(equal(1))
+                        expect((result?.errors.last as? MultiSaveError)?.code).to(equal(11000))
+                    }
+                    it("should return an entities and errors when some requests fail and some succeed") {
+                        mockResponse { request in
+                            switch request.httpMethod {
+                            case "POST":
+                                return HttpResponse(
+                                    statusCode: 207,
+                                    json: [
+                                        "entities": [
+                                            [
+                                                "name": "Victor",
+                                                "_acl": [
+                                                    "creator": UUID().uuidString
+                                                ],
+                                                "_kmd": [
+                                                    "lmt": Date().toISO8601(),
+                                                    "ect": Date().toISO8601()
+                                                ],
+                                                "_id": UUID().uuidString
+                                            ],
+                                            nil
+                                        ],
+                                        "errors" : [
+                                            [
+                                                "index" : 1,
+                                                "code": UUID().uuidString,
+                                                "errmsg": "Geolocation points must be in the form [longitude, latitude] with long between -180 and 180, lat between -90 and 90"
+                                            ]
+                                        ]
+                                    ]
+                                )
+                            default:
+                                return HttpResponse(request: request, urlProcotolType: KinveyURLProtocol.self)
+                            }
+                        }
+                        defer {
+                            setURLProtocol(nil)
+                        }
+                        let result = kinveySaveMulti(
+                            dataStore: syncDataStore,
+                            entities: [
+                                Person { $0.name = "Victor" },
+                                Person { $0.geolocation = GeoPoint(latitude: -300, longitude: -300
```

Comment:
we should check that the correct item is here - and the other items are missing

Reply:
we are checking, take a look in the next lines. You will see that we are checking the local cache should have 2 items and the network store returns only 1 item while the other item still to be synced as a pending entity

Reply:
The comment above regards this test. Why are the assertions below changed? The correct ones, I think, are 2 for network store and 3 for sync

Reply:
Victor, It would be best if this test follows the correct flow disregarding the bug in KCS that returns a single error instead of returning success for the 2 correct items and an error for the invalid one. And just skip it until the issue is fixed. So the correct behavior would be for sync to return success for items 0 and 2 and error for item 1. So the pending item would only be item 1. Also, in the file I forgot to name the test. I have added a name: "should handle multi-insert push error properly"

Reply:
we are now mocking the response the reproduce the situation described by the test

Reply:
I seem to have made a poor job of describing this test case. Here is how I see it:
1. Create 2 valid and 1 invalid item
2. Save, but mock connectivity error, so the items are not actually saved in the backend but are saved offline
3. Check if they are in the pendingSync queue
4. Sync - here we can mock the request because of the backend bug, so it returns success for the valid items and error for the invalid ones
5. Check the pendingSync queue that the invalid item is still there
6. Check the pendingSync queue that all items are there.